pkg/services/supportbundles/supportbundlesimpl/service_bundle.go

```diff
@@ -92,7 +95,33 @@ func (s *Service) bundle(ctx context.Context, collectors []string, uid string) (
 		return nil, errCompress
 	}
 
-	return buf.Bytes(), nil
+	final := buf
+	if len(s.encryptionPublicKeys) > 0 {
+		final = bytes.Buffer{}
+		recipients := make([]age.Recipient, 0, len(s.encryptionPublicKeys))
+		for _, key := range s.encryptionPublicKeys {
+			recipient, err := age.ParseX25519Recipient(key)
+			if err != nil {
+				return nil, fmt.Errorf("unable to parse support bundle recipient public key: %w", err)
+			}
+			recipients = append(recipients, recipient)
+		}
+
+		w, err := age.Encrypt(&final, recipients...)
+		if err != nil {
+			return nil, fmt.Errorf("unable to open support bundle encryption header: %w", err)
+		}
+
+		if _, err = w.Write(buf.Bytes()); err != nil {
+			return nil, fmt.Errorf("unable to write support bundle encryption: %w", err)
+		}
+
+		if err := w.Close(); err != nil {
+			return nil, fmt.Errorf("unable to close support bundle encryption: %w", err)
+		}
```

Comment:
given the non-trivial complexity here, and also the fact there's a `compress` function too, I wonder if extracting this to an `encrypt` function might be ideal 🤔